### PR TITLE
[release/3.1] Update WixVersion

### PR DESCRIPTION
**Pull request by wtgodbe**
Update to a newer version of WiX that includes a signed wixstdba.dll. Should resolve https://github.com/dotnet/aspnetcore/issues/18091

### CODE REVIEW

**Review by joeloff — APPROVED** [review]

**Review by joeloff — APPROVED** [review]

File: eng/targets/Wix.Common.props [hunk 1]
```diff
@@ -3,8 +3,8 @@
 
   <PropertyGroup>
     <SchemaVersion>2.0</SchemaVersion>
-    <ProductVersion>3.11</ProductVersion>
-    <WixVersion>3.11.1</WixVersion>
+    <ProductVersion>3.14</ProductVersion>
+    <WixVersion>3.14.0-dotnet</WixVersion>
```
Comment (dougbu) [inline eng/targets/Wix.Common.props:7]: As @joeloff said in #18091,
> we should give it a try

There's a very good chance this package will _not_ change whether the wixstdba.dll file is signed. This package is an old fork and not that far removed from 3.11.1. Don't make this change if it leaves the file unsigned.

**Review by dougbu — APPROVED** [review]
If it works, :shipit:

File: eng/targets/Wix.Common.props (hunk 1 above)
Comment (wtgodbe) [reply eng/targets/Wix.Common.props:7]: I already validated that our 5.0 bundles (that are built with this version of wix) have a signed wixstdba.dll, while our 3.1 bundles have an unsigned one. I'll run an internal build of this branch to grab the bundles & double check them as well.

Comment (wtgodbe) [reply eng/targets/Wix.Common.props:7]: https://dev.azure.com/dnceng/internal/_build/results?buildId=977745&view=results

Comment (wtgodbe) [reply eng/targets/Wix.Common.props:7]: Just confirmed that the wixsdtba.dll is signed in the .exes produced by this build

Comment (dougbu) [reply eng/targets/Wix.Common.props:7]: Excellent

**Pull request merged**